docs/articles/documentation/guides/basic-guides/select-page-elements.md

```diff
@@ -0,0 +1,542 @@
+    const osCount            = Selector('.column.col-2 label').count;
+    const submitButtonExists = Selector('#submit-button').exists;
+
+    await t
+        .expect(osCount).eql(3)
+        .expect(submitButtonExists).ok();
+});
+```
+
+Note that selector property getters are asynchronous.
+
+### Obtain Element State
+
+Selectors and promises returned by selectors expose API that allows you to obtain the state (size, position, classes, etc.) of the matching element.
+
+```js
+const headerText = await Selector('#header').textContent;
+```
+
+See the API reference in [DOMNodeState](../../reference/test-api/domnodestate.md).
+
+The API is asynchronous. Use `await` before methods and properties.
+
+**Example**
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page('http://devexpress.github.io/testcafe/example/');
+
+const windowsInput = Selector('#windows');
+
+test('Obtain Element State', async t => {
+    await t.click(windowsInput);
+
+    const windowsInputChecked = await windowsInput.checked; // returns true
+});
+```
+
+#### DOM Node Snapshot
+
+If you need to use an object's state multiple times in the test, you may want to get the object that contains [all the data](../../reference/test-api/domnodestate.md) in one turnaround to the client. To obtain this object (*DOM Node Snapshot*), call the selector with the `await` keyword:
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('DOM Node Snapshot', async t => {
+    const sliderHandle         = Selector('#slider').child('span');
+    const sliderHandleSnapshot = await sliderHandle();
+
+    console.log(sliderHandleSnapshot.hasClass('ui-slider-handle'));    // => true
+    console.log(sliderHandleSnapshot.childElementCount);               // => 0
+});
+```
+
+Note that if a selector matches several DOM nodes on the page, this asynchronous call returns the first node from the matched set.
+
+> TestCafe wait mechanisms do not work when you pass a snapshot's property to an [assertion](assert.md). To enable [Smart Assertion Query]assert.md#smart-assertion-query-mechanism), pass [selector's properties](../../reference/test-api/domnodestate.md#members-common-across-all-nodes) instead.
+
+### Define Action Targets
+
+You can pass selectors to [test actions](interact-with-the-page.md) to specify the action's target element.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+const label = Selector('#tried-section').child('label');
+
+test('My Test', async t => {
+    await t.click(label);
+});
+```
+
+If the selector matches multiple DOM nodes, the action is executed for the first node.
+
+Before an action is performed, TestCafe waits for the target element to appear in the DOM and become visible. If this does not happen within the [selector timeout](#selector-timeout), the test fails.
```

Comment:
"appear in the DOM and become visible" appear and become visible - this means the same

Reply:
Yes, but elements that present in the DOM are not necessarily displayed. So, 'appearing in the DOM' does not mean visibility. These are two different conditions: 

1. an element is in the DOM,
2. an element is visible.